src/bin/languageserver/mod.rs

```diff
@@ -1230,3 +1266,156 @@ fn loc_to_range(loc: &pt::Loc, file: &ast::File) -> Range {
 
     Range::new(start, end)
 }
+
+fn update_file_contents(
+    mut prev_content: String,
+    content_change: TextDocumentContentChangeEvent,
+) -> String {
+    if let Some(range) = content_change.range {
+        let start_line = range.start.line as usize;
+        let start_col = range.start.character as usize;
+        let end_line = range.end.line as usize;
+        let end_col = range.end.character as usize;
+
+        // Directly add the changes to the buffer when changes are present at the end of the file.
+        if start_line == prev_content.lines().count() {
+            prev_content.push_str(&content_change.text);
+            return prev_content;
+        }
+
+        let mut new = String::new();
```

Comment:
This variable deserves a better name.

Reply:
I decided to rename it to `new_content` in accordance with its counterpart `prev_content`